app/controllers/trainees/diversity/disclosures_controller.rb

```diff
@@ -33,17 +33,24 @@ def disclosure_params
         params.require(:diversities_disclosure_form).permit(*Diversities::DisclosureForm::FIELDS)
       end
 
-      def redirect_to_relevant_step
-        if @disclosure_form.diversity_disclosed?
-          redirect_to(edit_trainee_diversity_ethnic_group_path(trainee))
-        else
-          trainee.apply_application? ? redirect_to(page_tracker.last_origin_page_path) : redirect_to(trainee_diversity_confirm_path(trainee))
-        end
-      end
-
       def authorize_trainee
         authorize(trainee)
       end
+
+      def step_wizard
+        @step_wizard ||= Wizards::Diversities::StepWizard.new(trainee: trainee, page_tracker: page_tracker)
```

Comment:
Is this the same code in the other diversity controllers? Could it not go into the parent controller so it's one place? or at least `step_wizard` and `user_came_from_banklink?`

Reply:
step wizard is different (schools/diversities) but the `user_came_from_backlink` can definitely go up

Reply:
Also, the folder hierarchy - i think it can just be `/lib/wizards/diversity_step_wizard`. I don't think the extra name spacing is necessary at the moment.

Reply:
all done 👍